### DISCUSSION TITLE
[RESOLVED] A fatal error occurred: Cannot downgrade

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [github/codeql#8495] [General]
I'm testing the cpp queries, with this source code: [bug.cpp.txt](https://github.com/github/codeql/files/8308419/bug.cpp.txt)

Then I create the database:

```
codeql database create cpptest -l=cpp -c="g++ bug.cpp" --overwrite
```

It succeed but codeql v2.8.3 fails to run query: [log.txt](https://github.com/github/codeql/files/8308420/log.txt)

**Comment by tausbn** [comment]
That's very strange. A database you've just created (and with a new version of the CLI) shouldn't need downgrading.

Could it be that your CodeQL libraries are out of date? You may need to pull the latest changes from `github/codeql` (or whatever repo you got the queries from).

**Reply by CaledoniaProject** [reply]
%> codeql --version
CodeQL command-line toolchain release 2.8.3.
Copyright (C) 2019-2022 GitHub, Inc.

**Comment by CaledoniaProject** [comment]
Interesting, I've just updated the submodule of starter project, and the problem is gone
